tests/pyfunc/test_model_export_with_class_and_artifacts.py

```diff
@@ -2542,3 +2543,43 @@ def predict(self, model_input: list[str]) -> list[str]:
     assert len(traces) == 3
     for i in range(3):
         assert traces[i].info.request_metadata[TraceMetadataKey.MODEL_ID] == model_infos[i].model_id
+
+
+def test_lock_model_requirements(monkeypatch: pytest.MonkeyPatch, tmp_path: Path):
```

Comment:
Could we add more tests
- with constraints
- check specific library is locked (pandas, numpy)
- check this works with indexes (maybe manual test on databricks, and ensure this works on serverless)